### PR TITLE
Incorporate misc updates as per project audit

### LINKED ISSUE #134: Review and audit entire project
Take the time to revise HMN in its entirety, to clean up, eliminate inconsistencies, prepare for next features.

### PR DESCRIPTION
**Pull request by j9t**
Resolves #134

## Summary by CodeRabbit

* **New Features**
  * Enhanced demo UI and layout with live share and improved inputs/outputs.
  * README clarified support for in-document CSS, JS, and SVG minification.

* **Chores**
  * Released v4.17.0; Dependabot and CI workflow updates; benchmark configuration extended.

* **Bug Fixes**
  * Improved attribute/quote handling and smarter whitespace preservation around form controls and hidden inputs.

* **Tests**
  * Expanded and clarified test coverage for CSS/JS/JSON and related edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/lib/constants.js [hunk 1]
```diff
@@ -119,35 +125,35 @@ const optionalStartTags = new Set(['html', 'head', 'body', 'colgroup', 'tbody'])
...
...e', 'audio', 'b', 'base', 'basefont', 'bdi', 'bdo', 'bgsound', 'big', 'blink', 'blockquote', 'body', 'br', 'button', 'canvas', 'caption', 'center', 'cite', 'code', 'col', 'colgroup', 'command', 'content', 'data', 'datalist', 'dd', 'del', 'details', 'dfn', 'dialog', 'dir', 'div', 'dl', 'dt', 'element', 'em', 'embed', 'fieldset', 'figcaption', 'figure', 'font', 'footer', 'form', 'frame', 'frameset', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'head', 'header', 'hgroup', 'hr', 'html', 'i', 'iframe', 'image', 'img', 'input', 'ins', 'isindex', 'kbd', 'keygen', 'label', 'legend', 'li', 'link', 'listing', 'main', 'map', 'mark', 'marquee', 'menu', 'menuitem', 'meta', 'meter', 'multicol', 'nav', 'nobr', 'noembed', 'noframes', 'noscript', 'object', 'ol', 'optgroup', 'option', 'output', 'p', 'param', 'picture', 'plaintext', 'pre', 'progress', 'q', 'rb', 'rp', 'rt', 'rtc', 'ruby', 's', 'samp', 'script', 'search', 'section', 'select', 'selectedcontent', 'shadow', 'small', 'source', 'spacer', 'span', 'strike', 'strong', 'style', 'sub', 'summary', 'sup', 'table', 'tbody', 'td', 'template', 'textarea', 'tfoot', 'th', 'thead', 'time', 'title', 'tr', 'track', 'tt', 'u', 'ul', 'var', 'video', 'wbr', 'xmp']);
```
[the remark this answers is not in the document]
Comment (j9t) [reply src/lib/constants.js:156]: These constants are not part of the public API.
